### ISSUE TITLE
return Function_Stop; in onStartCountdown doesn’t prevent the music to play

### ISSUE THREAD

**Issue by marius851000** [ShadowMario/FNF-PsychEngine#10701]
### What is your question?

**Background (can be skipped, just like to write a lot of useless stuff)**

Hello. I have been working on packaging Psych Engine in my distro (nixpkgs). I had two issues (outside of the generic stuff more related to haxe than this project, including some problem regarding making cache work):
1. Video didn’t player
2. The engine require mods to be in the same folder that the installation folder (which is a problem due to the installation folder being a read-only filesystem, but I’ll work on patch for this later, and is not the matter of this question)

Regarding the video player, I already solved two issues, that make it able to actually start the video

(for later reference, these were: 1. enable `VIDEOS_ALLOWED` on linux, and ensure libvlc is compiled with the same version that `linc_luajit` is compiled with (or just disable lua in vlc, which I did until I find a better solution) due to symbol conflict)

I also have another issue, but that’s seemingly not an issue with FNF-PsychEngine (but if it’s of any use, it’s libvlc using the nvidia driver for hardware video decoding, where it should use radeon. VLC work well, hand I can force it to use Radeon, so I can workaround it temporarily)

**The problem itself**

 (using FNF 6.0.2 on Linux with video support enabled)

In the Dusk Till Dawn mod, the following piece of code is present:
```lua
function onStartCountdown()
    print(isStoryMode)
    print(seenCutscene)
    if not lowQuality then
        if not allowCountdown and isStoryMode and not seenCutscene then --Block the first countdown
            startVideo('DTD_Intro');
            print('video played!?');
            allowCountdown = true;
            return Function_Stop;
        end
        
    end
    return Function_Continue;
end
```

(print added by me)

It turns out that, while the video does play, the music still start playing (which isn’t supposed to happen). startVideo should eventually re-call `StartCountdown` when the video is finished (leading to branch with `return Function_Continue`).

From (abusive) trace-debugging, it appear the end-of-video callback isn’t called (at least, not until after around 20 second after the beggining, when the video actually end). It also seems that the main body of `StartCountdown` is not called (as `startedCountdown` is set to `true`), which cause further issues.

**Does someone can at least tell me if there is some update that broke compatiblity, or if there is something obviously wrong here?** I’ll try to continue to fix that in the meantime.

**Comment by marius851000** [comment]
correction StartedCountdown body’s IS called despite Function_Stop being returned (seemingly).

**Comment by marius851000** [comment]
Okay. I thing it’s an issue fixed on master, described here: https://github.com/ShadowMario/FNF-PsychEngine/blob/main/source/PlayState.hx#L5141

(comparing floats doesn’t seems to be a good idea)

**Comment by MemeHoovy** [comment]
> Okay. I thing it’s an issue fixed on master, described here: https://github.com/ShadowMario/FNF-PsychEngine/blob/main/source/PlayState.hx#L5141
> 
> (comparing floats doesn’t seems to be a good idea)

I mean, there is this same code done in a different way from a fork.
https://github.com/MemeHoovy/FNF-PsychEngine-Extra-1/blob/main/source/PlayState.hx#L6508

**Comment by MemeHoovy** [comment]
Not sure if it's any better or any worse

**Comment by marius851000** [comment]
Well... Further looks and experiment seems to be another thing. Like lua seemingly returning 1.03753785626662e-322 instead of 1.0000000(something)e0

**Comment by MemeHoovy** [comment]
> Well... Further looks and experiment seems to be another thing. Like lua seemingly returning 1.03753785626662e-322 instead of 1.0000000(something)e0

basically a long number like PI?

**Comment by marius851000** [comment]
> > Well... Further looks and experiment seems to be another thing. Like lua seemingly returning 1.03753785626662e-322 instead of 1.0000000(something)e0
> 
> basically a long number like PI?

0x0000000000000001 as 64-bit float (if I’m correct)

**Comment by marius851000** [comment]
*and it actually work on main-1 (main has some issue related to pitch something). Probably this comparaison or something. I planned to rewrite that use int instead of Dynamic, but seeing the time I spent on it and that it work, I’ll not question it more*

**Comment by MemeHoovy** [comment]
> _and it actually work on main-1 (main has some issue related to pitch something). Probably this comparaison or something. I planned to rewrite that use int instead of Dynamic, but seeing the time I spent on it and that it work, I’ll not question it more_

I'm not getting what your saying

**Comment by CoolingTool** [comment]
> 

yeah this fork has said issue

**Comment by MemeHoovy** [comment]
> > I mean, there is this same code done in a different way from a fork.
> > https://github.com/MemeHoovy/FNF-PsychEngine-Extra-1/blob/main/source/PlayState.hx#L6508
> 
> yeah this fork has said issue

figured starmapo fixed it but apparently not

**Comment by marius851000** [comment]
> > _and it actually work on main-1 (main has some issue related to pitch something). Probably this comparaison or something. I planned to rewrite that use int instead of Dynamic, but seeing the time I spent on it and that it work, I’ll not question it more_
> 
> I'm not getting what your saying

Version 0.6.2 doesn’t work, but ff917e6b1de454d27dc7d95c7169e41ff9154ee8 work (and version 1ccd799fc9f0b567bf92602c79adf951cbfe1d81 refused to compile for an unrelated reason when I wrote this, althought it has seemingly been fixed later with)

**Comment by CoolingTool** [comment]
so i happened to be rereading this issue then noticed
> (for later reference, these were: 1. enable VIDEOS_ALLOWED on linux, and ensure libvlc is compiled with the same version that linc_luajit is compiled with (or just disable lua in vlc, which I did until I find a better solution) due to symbol conflict)

i had no idea this was the reason why i could not get videos to work on linux
i installed vlc-luajit from the arch user repository and surprise it worked
thanks alot man
